PsfShimMonitor/MainWindow.xaml.cs

```diff
@@ -0,0 +1,720 @@
+                        catch { }
+                    }
+                    try
+                    {
+                        start = (Int64)data.PayloadByName("Start");
+                    }
+                    catch { }
+                    try
+                    {
+                        end = (Int64)data.PayloadByName("End");
+                    }
+                    catch { }
+                    EventItem ei = new EventItem((int)data.EventIndex,
+                                                    start,
+                                                    end,
+                                                    data.TimeStamp,
+                                                    data.ProcessName,  
+                                                    data.ProcessID,
+                                                    data.ThreadID,
+                                                    data.ProviderName,
+                                                    operation,
+                                                    inputs,
+                                                    result,
+                                                    outputs,
+                                                    caller
+                                                  );
+
+                    lock (_TEventListsLock)
+                    {
+                        _TEventListItems.Add(ei);
+                        AddToProcIDsList(data.ProcessID);
+                    }
+                    worker.ReportProgress((int)data.EventIndex);
+                };
+
+                res1 = myTraceEventSession.EnableProvider(etwp.guid);    
+                if (!res1)
+                {
+                    // Attempt resetting for second run...
+                    myTraceEventSession.DisableProvider(etwp.guid);
+                    res1 = myTraceEventSession.EnableProvider(etwp.guid);
+                }
+                res2 = myTraceEventSession.Source.Process();
+            }
+        } // Eventbgw_DoWork()
+        private void AddToProcIDsList(int pid)
+        {
+            foreach (int match in ProcIDsOfTarget)
+            {
+                if (match == pid) return;
+            }
+            ProcIDsOfTarget.Add(pid);
+        }
+        private bool IsPidInProdIDsList(int pid)
+        {
+            foreach (int match in ProcIDsOfTarget)
+            {
+                if (match == pid) return true;
+            }
+            return false;
+        }
+        private void Eventbgw_ProgressChanged(object sender, ProgressChangedEventArgs e)
+        {
+            try
+            {
+                lock (_TEventListsLock)
+                {
+                    if (_TEventListItems.Count > 0)
+                    {
+                        foreach (EventItem ei in _TEventListItems)
+                        {
+                            if (FilterOnProcessId < 0)
+                                FilterOnProcessId = ei.ProcessID;
```

Comment:
Please add brackets to all single line if/else's.  They help improve readability and prevents errors in future edits.

Reply:
whatever. done.

Reply:
Thank you.